runner.sh

```diff
@@ -1,25 +1,85 @@
-dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-cd $dir
+SRC_ROOT=$(dirname $1)
+PLUGIN_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-# Clean all for now. Hack to display errors, and ideally
-# we will not do this!
-rm -rf build
+# Derive a tmp build directory
+BUILD_DIR="/tmp/SwiftPlayground-$( echo $SRC_ROOT | md5 )"
 
-mkdir -p build
+# Remove if it already existed. Errors seem to be swallowed otherwise
+rm -rf $BUILD_DIR
+
+mkdir -p $BUILD_DIR
+cd $BUILD_DIR
 
 # For now, dump the entire file into main this is likely contents.swift
-cat $1 > build/main.swift
-
-xcrun swiftc \
--Xfrontend \
--playground \
--Xfrontend \
--debugger-support \
--module-name Playgound \
--o \
-build/main \
-build/main.swift \
-PlaygroundRuntime.swift
-
-./build/main
+# Consider writing the buffer instead
+cat $1 > main.swift
+
+# Build up a SwiftC invocation and then run
+
+if [[ -d $SRC_ROOT/Sources/ ]]; then
+    # Framework includes, rpath, and source includes
+    PLAYGROUND_FLAGS="-F$SRC_ROOT/Sources/ -I$SRC_ROOT/Sources/ -Xlinker -rpath -Xlinker $SRC_ROOT/Sources/"
+
+    # WARNING! this may be dangerous in some terminals.
+    PLAYGROUND_SOURCES="$( find $SRC_ROOT/Sources/*.swift )"
+else
+    PLAYGROUND_FLAGS=""
+    PLAYGROUND_SOURCES=""
+fi
+
+# Check contents.xcplayground for ios
+grep -q "target-platform='ios'" "$SRC_ROOT/contents.xcplayground"
+if [[ $? == 0 ]]; then
+    # Build and run for iOS
+    xcrun swiftc \
+    -Xfrontend \
+    -debugger-support \
+    -Xfrontend \
+    -playground \
+    -module-name \
+    Playground \
+    -target \
+    x86_64-apple-ios7.0 \
+    -sdk \
+    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk \
+    -F \
+    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks \
+    -F \
+    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks \
+    -F \
+    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks \
+    -F \
+    /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/PrivateFrameworks \
```

Comment:
Might want to use `xcode-select -p` here instead so it works as users change xcode versions